From 5654f4d2f90f95efd1f0ca70b438a3ab81022d15 Mon Sep 17 00:00:00 2001
From: Frediano Ziglio <fziglio@redhat.com>
Date: Thu, 26 Mar 2020 11:31:50 +0000
Subject: [PATCH] systemd-login: Avoid a crash on container

References: bsc#1173749

On containers dbus could be not running.
In this case dbus.system_connection is NULL and calling
dbus_connection_close on it will cause a crash.
This happens also under Gitlab CI.

Signed-off-by: Frediano Ziglio <fziglio@redhat.com>
Acked-by: Victor Toso <victortoso@redhat.com>
---
 src/vdagentd/systemd-login.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/vdagentd/systemd-login.c b/src/vdagentd/systemd-login.c
index 0b8f3c1..2d2311c 100644
--- a/src/vdagentd/systemd-login.c
+++ b/src/vdagentd/systemd-login.c
@@ -250,7 +250,9 @@ void session_info_destroy(struct session_info *si)
         return;
 
     si_dbus_match_remove(si);
-    dbus_connection_close(si->dbus.system_connection);
+    if (si->dbus.system_connection) {
+        dbus_connection_close(si->dbus.system_connection);
+    }
     sd_login_monitor_unref(si->mon);
     g_free(si->session);
     g_free(si);
-- 
2.29.0

